src/solc_0.5/Land/erc721/LandBaseTokenV3.sol

```diff
@@ -39,6 +39,7 @@ contract LandBaseTokenV3 is ERC721BaseTokenV2 {
         bytes calldata data
     ) external {
         require(to != address(0), "to is zero address");
+        require(size != 0, "size cannot be zero");
```

Comment:
the size is actually already checked in `_getQuadLayer`
so it's a bit wasted gas here

Reply:
maybe you could do like you did on the 0.8 version: 
merge all the require about size into one function